Could not complile under FreeBSD 10.3, Clang 3.6.2 and 3.8.0: boost error is "call to 'f' is ambiguous"

FreeBSD 10.3, Clang 3.6.2
FreeBSD 10.3, Clang 3.8.0

Detail log:

```
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:19:
In file included from /usr/local/include/boost/mpl/sequence_tag.hpp:18:
In file included from /usr/local/include/boost/mpl/aux_/has_tag.hpp:17:
In file included from /usr/local/include/boost/mpl/has_xxx.hpp:19:
In file included from /usr/local/include/boost/mpl/bool.hpp:17:
In file included from /usr/local/include/boost/mpl/bool_fwd.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/adl_barrier.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/config/adl.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/config/workaround.hpp:17:
/usr/local/include/boost/detail/workaround.hpp:44:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __BORLANDC___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:49:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __CODEGEARC___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:54:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _MSC_VER_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:59:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _MSC_FULL_VER_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:76:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __GNUC___WORKAROUND_GUARD 0
        ^
/usr/local/include/boost/detail/workaround.hpp:81:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __GNUC_MINOR___WORKAROUND_GUARD 0
        ^
/usr/local/include/boost/detail/workaround.hpp:86:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __GNUC_PATCHLEVEL___WORKAROUND_GUARD 0
        ^
/usr/local/include/boost/detail/workaround.hpp:89:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __IBMCPP___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:94:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __SUNPRO_CC_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:99:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __DECCXX_VER_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:104:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __MWERKS___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:109:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __EDG___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:114:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __EDG_VERSION___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:119:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __HP_aCC_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:124:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __hpxstd98_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:129:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _CRAYC_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:134:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __DMC___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:144:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __COMO___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:149:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __COMO_VERSION___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:154:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __INTEL_COMPILER_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:159:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __ICL_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:164:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _COMPILER_VERSION_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:170:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _RWSTD_VER_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:180:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __GLIBCPP___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:185:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _GLIBCXX_USE_C99_FP_MACROS_DYNAMIC_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:190:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __SGI_STL_PORT_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:195:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _STLPORT_VERSION_WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:200:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define __LIBCOMO_VERSION___WORKAROUND_GUARD 1
        ^
/usr/local/include/boost/detail/workaround.hpp:205:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _CPPLIB_VER_WORKAROUND_GUARD 1
        ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:27:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:45:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:63:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:81:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:99:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:1:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/attribute.hpp:8:
In file included from /usr/local/include/boost/mpl/contains.hpp:20:
In file included from /usr/local/include/boost/mpl/aux_/contains_impl.hpp:20:
In file included from /usr/local/include/boost/mpl/find.hpp:17:
In file included from /usr/local/include/boost/mpl/find_if.hpp:17:
In file included from /usr/local/include/boost/mpl/aux_/find_if_pred.hpp:14:
In file included from /usr/local/include/boost/mpl/aux_/iter_apply.hpp:17:
In file included from /usr/local/include/boost/mpl/apply.hpp:24:
In file included from /usr/local/include/boost/mpl/placeholders.hpp:24:
In file included from /usr/local/include/boost/mpl/arg.hpp:37:
In file included from /usr/local/include/boost/mpl/aux_/include_preprocessed.hpp:37:
/usr/local/include/boost/mpl/aux_/preprocessed/gcc/arg.hpp:117:9: warning: use of old-style cast [-Wold-style-cast]
        BOOST_MPL_AUX_ASSERT_NOT_NA(type);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/aux_/na_assert.hpp:25:5: note: expanded from macro 'BOOST_MPL_AUX_ASSERT_NOT_NA'
    BOOST_MPL_ASSERT_NOT((boost::mpl::is_na<type>)) \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/mpl/assert.hpp:312:43: note: expanded from macro 'BOOST_MPL_ASSERT_NOT'
              boost::mpl::assert_not_arg( (void (*) pred)0, 1 ) \
                                          ^              ~
/usr/local/include/boost/mpl/assert.hpp:59:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:7:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/detail/attribute.hpp:3:
In file included from /usr/local/include/boost/variant/variant.hpp:32:
/usr/local/include/boost/variant/detail/forced_return.hpp:53:1: warning: function 'forced_return<void>' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
{
^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:7:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/detail/attribute.hpp:3:
In file included from /usr/local/include/boost/variant/variant.hpp:33:
In file included from /usr/local/include/boost/variant/detail/initializer.hpp:23:
In file included from /usr/local/include/boost/variant/detail/move.hpp:27:
In file included from /usr/local/include/boost/move/move.hpp:23:
In file included from /usr/local/include/boost/move/traits.hpp:19:
In file included from /usr/local/include/boost/type_traits/is_nothrow_move_constructible.hpp:21:
In file included from /usr/local/include/boost/utility/declval.hpp:15:
/usr/local/include/boost/type_traits/add_rvalue_reference.hpp:9:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define BOOST_TYPE_TRAITS_EXT_ADD_RVALUE_REFERENCE__HPP
        ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:5:
In file included from /usr/local/include/boost/variant/get.hpp:21:
/usr/local/include/boost/utility/addressof.hpp:59:12: error: call to 'f' is ambiguous
    return boost::detail::addressof_impl<T>::f( boost::detail::addr_impl_ref<T>( v ), 0 );
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/variant/get.hpp:85:23: note: in instantiation of function template specialization 'boost::addressof<const nullptr_t>' requested here
        return boost::addressof(operand);
                      ^
/usr/local/include/boost/variant/variant.hpp:1048:16: note: in instantiation of member function 'boost::detail::variant::get_visitor<const nullptr_t>::operator()' requested here
        return visitor_(operand);
               ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:128:20: note: in instantiation of function template specialization 'boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> >::internal_visit<const nullptr_t>' requested here
    return visitor.internal_visit(
                   ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:170:13: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl_invoke_impl<boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> >, const void *, nullptr_t>' requested here
    return (visitation_impl_invoke_impl)(
            ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:258:11: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl_invoke<boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> >, const void *, nullptr_t, boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>' requested here
        , BOOST_VARIANT_AUX_APPLY_VISITOR_STEP_CASE
          ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:38:60: note: expanded from macro 'BOOST_PP_REPEAT_1'
# define BOOST_PP_REPEAT_1(c, m, d) BOOST_PP_REPEAT_1_I(c, m, d)
                                                           ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:43:63: note: expanded from macro 'BOOST_PP_REPEAT_1_I'
# define BOOST_PP_REPEAT_1_I(c, m, d) BOOST_PP_REPEAT_1_ ## c(m, d)
                                                              ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:71:58: note: expanded from macro 'BOOST_PP_REPEAT_1_20'
# define BOOST_PP_REPEAT_1_20(m, d) BOOST_PP_REPEAT_1_19(m, d) m(2, 19, d)
                                                         ^
note: (skipping 17 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:53:56: note: expanded from macro 'BOOST_PP_REPEAT_1_2'
# define BOOST_PP_REPEAT_1_2(m, d) BOOST_PP_REPEAT_1_1(m, d) m(2, 1, d)
                                                       ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:52:36: note: expanded from macro 'BOOST_PP_REPEAT_1_1'
# define BOOST_PP_REPEAT_1_1(m, d) m(2, 0, d)
                                   ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:249:17: note: expanded from macro 'BOOST_VARIANT_AUX_APPLY_VISITOR_STEP_CASE'
        return (visitation_impl_invoke)( \
                ^
/usr/local/include/boost/variant/variant.hpp:2362:33: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl<mpl_::int_<0>, boost::detail::variant::visitation_impl_step<boost::mpl::v_iter<boost::mpl::v_item<std::__1::function<void (blackhole::v1::writer_t &)>, boost::mpl::v_item<std::__1::basic_string<char>, boost::mpl::v_item<double, boost::mpl::v_item<unsigned long, boost::mpl::v_item<long, boost::mpl::v_item<bool, boost::mpl::v_item<nullptr_t, boost::mpl::vector0<mpl_::na>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, boost::mpl::v_iter<boost::mpl::v_item<std::__1::function<void (blackhole::v1::writer_t &)>, boost::mpl::v_item<std::__1::basic_string<char>, boost::mpl::v_item<double, boost::mpl::v_item<unsigned long, boost::mpl::v_item<long, boost::mpl::v_item<bool, boost::mpl::v_item<nullptr_t, boost::mpl::vector0<mpl_::na>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, 7> >, boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> >, const void *, boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>' requested here
        return detail::variant::visitation_impl(
                                ^
/usr/local/include/boost/variant/variant.hpp:2387:16: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor_impl<boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> >, const void *>' requested here
        return internal_apply_visitor_impl(
               ^
/usr/local/include/boost/variant/variant.hpp:2411:22: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor<boost::detail::variant::invoke_visitor<boost::detail::variant::get_visitor<const nullptr_t> > >' requested here
        return this->internal_apply_visitor(invoker);
                     ^
/usr/local/include/boost/variant/get.hpp:166:21: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::apply_visitor<boost::detail::variant::get_visitor<const nullptr_t> >' requested here
    return operand->apply_visitor(v);
                    ^
/home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:156:30: note: in instantiation of function template specialization 'boost::get<nullptr_t, boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, std::__1::basic_string<char>, std::__1::function<void (blackhole::v1::writer_t &)>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>' requested here
    if (auto result = boost::get<T>(&value.inner().value)) {
                             ^
/usr/local/include/boost/utility/addressof.hpp:37:23: note: candidate function
    static inline T * f( T & v, long )
                      ^
/usr/local/include/boost/utility/addressof.hpp:43:23: note: candidate function
    static inline T * f( T * v, int )
                      ^
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:7:
In file included from /home/username/blackhole2/blackhole-1.0.0-0alpha8/include/blackhole/detail/attribute.hpp:3:
/usr/local/include/boost/variant/variant.hpp:997:20: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
        return lhs == rhs;
               ~~~ ^  ~~~
/usr/local/include/boost/variant/variant.hpp:979:16: note: in instantiation of function template specialization 'boost::detail::variant::equal_comp::operator()<double>' requested here
        return Comp()(lhs_content, rhs_content);
               ^
/usr/local/include/boost/variant/variant.hpp:1048:16: note: in instantiation of function template specialization 'boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp>::operator()<double>' requested here
        return visitor_(operand);
               ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:128:20: note: in instantiation of function template specialization 'boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >::internal_visit<const double>' requested here
    return visitor.internal_visit(
                   ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:170:13: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl_invoke_impl<boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >, const void *, double>' requested here
    return (visitation_impl_invoke_impl)(
            ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:258:11: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl_invoke<boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >, const void *, double, boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>' requested here
        , BOOST_VARIANT_AUX_APPLY_VISITOR_STEP_CASE
          ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:38:60: note: expanded from macro 'BOOST_PP_REPEAT_1'
# define BOOST_PP_REPEAT_1(c, m, d) BOOST_PP_REPEAT_1_I(c, m, d)
                                                           ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:43:63: note: expanded from macro 'BOOST_PP_REPEAT_1_I'
# define BOOST_PP_REPEAT_1_I(c, m, d) BOOST_PP_REPEAT_1_ ## c(m, d)
                                                              ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:71:58: note: expanded from macro 'BOOST_PP_REPEAT_1_20'
# define BOOST_PP_REPEAT_1_20(m, d) BOOST_PP_REPEAT_1_19(m, d) m(2, 19, d)
                                                         ^
note: (skipping 13 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:57:56: note: expanded from macro 'BOOST_PP_REPEAT_1_6'
# define BOOST_PP_REPEAT_1_6(m, d) BOOST_PP_REPEAT_1_5(m, d) m(2, 5, d)
                                                       ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:56:62: note: expanded from macro 'BOOST_PP_REPEAT_1_5'
# define BOOST_PP_REPEAT_1_5(m, d) BOOST_PP_REPEAT_1_4(m, d) m(2, 4, d)
                                                             ^
/usr/local/include/boost/variant/detail/visitation_impl.hpp:249:17: note: expanded from macro 'BOOST_VARIANT_AUX_APPLY_VISITOR_STEP_CASE'
        return (visitation_impl_invoke)( \
                ^
/usr/local/include/boost/variant/variant.hpp:2362:33: note: in instantiation of function template specialization 'boost::detail::variant::visitation_impl<mpl_::int_<0>, boost::detail::variant::visitation_impl_step<boost::mpl::v_iter<boost::mpl::v_item<blackhole::v1::attribute::view_t::function_type, boost::mpl::v_item<blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, boost::mpl::v_item<double, boost::mpl::v_item<unsigned long, boost::mpl::v_item<long, boost::mpl::v_item<bool, boost::mpl::v_item<nullptr_t, boost::mpl::vector0<mpl_::na>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, boost::mpl::v_iter<boost::mpl::v_item<blackhole::v1::attribute::view_t::function_type, boost::mpl::v_item<blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, boost::mpl::v_item<double, boost::mpl::v_item<unsigned long, boost::mpl::v_item<long, boost::mpl::v_item<bool, boost::mpl::v_item<nullptr_t, boost::mpl::vector0<mpl_::na>, 0>, 0>, 0>, 0>, 0>, 0>, 0>, 7> >, boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >, const void *, boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>' requested here
        return detail::variant::visitation_impl(
                                ^
/usr/local/include/boost/variant/variant.hpp:2387:16: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor_impl<boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >, const void *>' requested here
        return internal_apply_visitor_impl(
               ^
/usr/local/include/boost/variant/variant.hpp:2411:22: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor<boost::detail::variant::invoke_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> > >' requested here
        return this->internal_apply_visitor(invoker);
                     ^
/usr/local/include/boost/variant/variant.hpp:2310:20: note: in instantiation of function template specialization 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::apply_visitor<boost::detail::variant::comparer<boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, boost::detail::variant::equal_comp> >' requested here
        return rhs.apply_visitor(visitor);
                   ^
/home/username/blackhole2/blackhole-1.0.0-0alpha8/src/attribute.cpp:247:26: note: in instantiation of member function 'boost::variant<boost::detail::variant::over_sequence<boost::mpl::vector<nullptr_t, bool, long, unsigned long, double, blackhole::v1::cpp17::basic_string_view<char, std::__1::char_traits<char> >, blackhole::v1::attribute::view_t::function_type, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na> >, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::operator==' requested here
    return inner().value == other.inner().value;
                         ^
38 warnings and 1 error generated.

                 ^

```

--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+blackhole (1.0.0-0alpha14) unstable; urgency=low
+
+  * Bug fix: fix compilation on clang with boost 1.55.
+
+ -- Evgeny Safronov <division494@gmail.com>  Tue, 31 May 2016 22:50:35 +0300
+
 blackhole (1.0.0-0alpha13) unstable; urgency=low
 
   * Feature: meet the syslog sink.

--- a/src/attribute.cpp
+++ b/src/attribute.cpp
@@ -2,6 +2,8 @@
 
 #include <boost/type_traits/remove_cv.hpp>
 #include <boost/variant/apply_visitor.hpp>
+// Must be included strictly before <boost/variant/get.hpp>.
+#include "hack/addressof.hpp"
 #include <boost/variant/get.hpp>
 
 #include "blackhole/detail/attribute.hpp"

--- a/src/hack/addressof.hpp
+++ b/src/hack/addressof.hpp
@@ -1,0 +1,70 @@
+#pragma once
+
+#include <cstddef>
+
+#include <boost/config.hpp>
+#include <boost/version.hpp>
+#include <boost/utility/addressof.hpp>
+
+//! Workaround (honestly, a hack) for cases when Blackhole is being compiled with clang on systems
+//! with boost 1.55 on board.
+//!
+//! Stolen from https://svn.boost.org/trac/boost/ticket/5487.
+//!
+//! \note must be included before <boost/variant/get.hpp> or <boost/utility/addressof.hpp>.
+
+#if defined(__clang__) && (BOOST_VERSION / 100000 == 1 && BOOST_VERSION / 100 % 1000 == 55)
+
+#ifndef BOOST_NO_CXX11_NULLPTR
+
+namespace boost {
+namespace detail {
+
+#if defined(__clang__) && !defined(_LIBCPP_VERSION) && !defined(BOOST_NO_CXX11_DECLTYPE)
+    typedef decltype(nullptr) addr_nullptr_t;
+#else
+    typedef std::nullptr_t addr_nullptr_t;
+#endif
+
+template<>
+struct addressof_impl<addr_nullptr_t> {
+    typedef addr_nullptr_t T;
+
+    constexpr static T* f(T& v, int) {
+        return &v;
+    }
+};
+
+template<>
+struct addressof_impl<addr_nullptr_t const> {
+    typedef addr_nullptr_t const T;
+
+    constexpr static T* f(T& v, int) {
+        return &v;
+    }
+};
+
+template<>
+struct addressof_impl<addr_nullptr_t volatile> {
+    typedef addr_nullptr_t volatile T;
+
+    constexpr static T* f(T& v, int) {
+        return &v;
+    }
+};
+
+template<>
+struct addressof_impl<addr_nullptr_t const volatile> {
+    typedef addr_nullptr_t const volatile T;
+
+    constexpr static T* f( T& v, int) {
+        return &v;
+    }
+};
+
+}  // namespace detail
+}  // namespace boost
+
+#endif // BOOST_NO_CXX11_NULLPTR
+
+#endif

